ext/io/console/extconf.rb

```diff
@@ -5,6 +5,10 @@
 # See https://bugs.ruby-lang.org/issues/20345
 MakeMakefile::RbConfig ||= ::RbConfig
 
+have_func("rb_syserr_fail_str(0, Qnil)") or
```

Comment:
FWIW I think `have_func("rb_syserr_fail_str")` should be enough, both eventually error with on TruffleRuby 24.1.1:
```
conftest.c:15: undefined reference to `rb_syserr_fail_str'
```
for me on Linux, and both are `yes` on CRuby 3.2.2.